src/Features/Core/Portable/EditAndContinue/EditSession.cs

```diff
@@ -374,100 +383,188 @@ private AsyncLazy<DocumentAnalysisResults> GetDocumentAnalysisNoLock(Document do
                 },
                 cacheResult: true);
 
-            _analyses[document.Id] = new Analysis(document, lazyResults);
+            _analyses[document.Id] = (document, lazyResults);
             return lazyResults;
         }
 
-        internal ImmutableArray<DocumentId> GetDocumentsWithReportedRudeEdits()
+        internal ImmutableArray<DocumentId> GetDocumentsWithReportedDiagnostics()
+        {
+            lock (_documentsWithReportedDiagnosticsGuard)
+            {
+                return ImmutableArray.CreateRange(_documentsWithReportedDiagnostics);
+            }
+        }
+
+        internal void TrackDocumentWithReportedDiagnostics(DocumentId documentId)
         {
-            lock (_documentsWithReportedRudeEditsGuard)
+            lock (_documentsWithReportedDiagnosticsGuard)
             {
-                return ImmutableArray.CreateRange(_documentsWithReportedRudeEdits);
+                _documentsWithReportedDiagnostics.Add(documentId);
             }
         }
 
-        public async Task<ProjectAnalysisSummary> GetProjectAnalysisSummaryAsync(Project project, CancellationToken cancellationToken)
+        public async Task<SolutionUpdateStatus> GetSolutionUpdateStatusAsync(Solution solution, string sourceFilePath, CancellationToken cancellationToken)
         {
             try
             {
-                var baseProject = _baseSolution.GetProject(project.Id);
+                // TODO: paralellize
```

Comment:
💡 This comment isn't especially helpful. If this would be significant, we can file an issue. Otherwise, we can simply omit the comment and address it if in the future it ever shows up in performance reports.